resources/web/wwi/WebotsView.js

```diff
@@ -408,6 +414,40 @@ export default class WebotsView extends HTMLElement {
     }
   }
 
+  #queryNeedRobot(protoUrl) {
+    let dbUrl = window.location.href;
+    if (!dbUrl.includes('webots.cloud'))
+      dbUrl = 'https://testing.webots.cloud/';
+    dbUrl = new URL(dbUrl).hostname;
+
+    if (protoUrl.includes('raw.githubusercontent.com')) {
+      protoUrl = protoUrl.replace('raw.githubusercontent.com', 'github.com');
+      protoUrl = protoUrl.split('/');
+      protoUrl.splice(5, 0, 'blob');
+      protoUrl = protoUrl.join('/');
```

Comment:
some of these can probably be chained

Reply:
I could chained one, but as the splice directly apply on the string and does not return anything, I cannot chain further